### PR TITLE
Elasticsearch Package

### PR DESCRIPTION
**Pull request by sayden**
This PR migrates the elasticsearch filebeat and metricbeat module into a package.

I'll open it for review but I wouldn't be surprised if some thing aren't totally correct because the migration process for this particular package was very error prone with many manual steps.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.

### Reviewers checklist
The migration of this package required some big amount of custom stuff, I'm gonna write it here:
* This module had a huge amount of aliases at the module level. Those that weren't relevant for a particular metricset have been  manually removed.
* Aliases are related to Stack Monitoring UI stuff so they have been removed from the docs as they are irrelevant for users. The downside is that I had to manually write them in the `_dev/build/docs/README.md` to avoid that they are rewritten during the build proccess. This means they'll need to be updated manually.
* All references to metricbeat, filebeat, metricsets and filesets have been removed from the docs

## Screenshots

image

image

image

### CODE REVIEW

File: packages/elasticsearch/manifest.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  - name: elasticsearch
+    title: Elasticsearch logs and metrics
+    description: Collect logs and metrics from Elasticsearch instances
+    inputs:
+      - type: logfile
+        title: 'Collect Elasticsearch audit, deprecation, gc, server and slowlog logs (input: logfile)'
+        description: 'Collecting audit, deprecation, gc, server and slowlog logs from Elasticsearch instances (input: logfile)'
+      - type: elasticsearch/metrics
+        title: Collect Elasticsearch ccr, cluster_stats, enrich, index, index_recovery, index_summary, ml_job, node, node_stats, pending_tasks and shard metrics
+        description: Collecting ccr, cluster_stats, enrich, index, index_recovery, index_summary, ml_job, node, node_stats, pending_tasks and shard metrics from Elasticsearch instances
```
Comment (mtojek) [inline packages/elasticsearch/manifest.yml:27]: I suggest to transform it to a more human readable form and rephrase titles and descriptions :)

File: packages/elasticsearch/manifest.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+icons:
+  - src: /img/logo_elasticsearch.svg
+    title: logo elasticsearch
+    size: 32x32
+    type: image/svg+xml
+format_version: 1.0.0
+license: basic
+categories: ["elastic_stack"]
+conditions:
+  kibana.version: ^7.10.0
```
Comment (mtojek) [inline packages/elasticsearch/manifest.yml:16]: Please bump up the constraint to 7.15. We don't want this package to be available for earlier versions.

File: packages/elasticsearch/manifest.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+name: elasticsearch
+title: Elasticsearch
+version: 0.1.0
+release: beta
```
Comment (mtojek) [inline packages/elasticsearch/manifest.yml:4]: maybe mark it as experimental as it's still early stage and lot of stuff to do.

**Review by mtojek — APPROVED** [review]
Left few nit-picks to fine-tune. I guess you can pull it in and add more changes/tests in iterations.

File: packages/elasticsearch/manifest.yml (hunk 1 above)
Comment (sayden) [reply packages/elasticsearch/manifest.yml]: I left simple titles now, like "Collect elasticsearch logs" or "Collect elasticsearch metrics". The description has a longer text but still human readable and I wrote some descriptions on the data_streams themselves :+1:

File: packages/elasticsearch/manifest.yml (hunk 2 above)
Comment (sayden) [reply packages/elasticsearch/manifest.yml]: Okay. I wrote `^7.15.0` now, I really didn't know which was is the correct.

File: packages/elasticsearch/manifest.yml (hunk 3 above)
Comment (sayden) [reply packages/elasticsearch/manifest.yml]: Oh, I read somewhere that all packages must be released as beat and that's why I changed it from experimental to beta. No prob, I revert it back to experimental.

**Comment by sayden** [comment]
I have also updated the screenshot to show the changes requested

File: packages/elasticsearch/data_stream/index_recovery/manifest.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+type: metrics
+title: Elasticsearch index_recovery metrics
```
Comment (mtojek) [inline packages/elasticsearch/data_stream/index_recovery/manifest.yml:2]: nit: index recovery

File: packages/elasticsearch/data_stream/cluster_stats/manifest.yml [hunk 5]
```diff
@@ -0,0 +1,7 @@
+type: metrics
+title: Elasticsearch cluster_stats metrics
```
Comment (mtojek) [inline packages/elasticsearch/data_stream/cluster_stats/manifest.yml:2]: nit: cluster stats

**Review by mtojek — APPROVED** [review]
A couple of nit picks

BTW I saw that every data stream starts with "Elasticsearch ..." prefix. I'm not sure if it isn't redundant as all these data streams are only related to Elasticsearch.

**Comment by sayden** [comment]
> A couple of nit picks
> 
> BTW I saw that every data stream starts with "Elasticsearch ..." prefix. I'm not sure if it isn't redundant as all these data streams are only related to Elasticsearch.

Good point I was mostly following the naming of other modules I took as example.

**Pull request merged**